### PR TITLE
Remove kubecost from eks-prow-build cluster

### PR DESCRIPTION
**Pull request by koksay**
Based on #7226

### CODE REVIEW

File: infra/aws/terraform/prow-build-cluster/resources/kubecost/README.md [hunk 1]
```diff
@@ -1,120 +0,0 @@
...
-> [!IMPORTANT]
-> The `federated-store.yaml` must be updated with the correct AWS credentials
-
-<details>
-  <summary>federated-store.yaml</summary>
-
-  ```yaml
-  type: S3
-  config:
-    bucket: "kubecost-prow-clusters-metrics"
```
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/resources/kubecost/README.md:68]: @koksay Should we also delete this bucket?

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

File: infra/aws/terraform/prow-build-cluster/resources/kubecost/README.md (hunk 1 above)
Comment (koksay) [reply infra/aws/terraform/prow-build-cluster/resources/kubecost/README.md:68]: Yes, I had done a couple of manual steps, I will revert them all.

**Pull request merged**